Scripts/GameSaver.gd

```diff
@@ -52,54 +56,101 @@ func load_game():
 				"_golds_on_second":
-					game_data.golds_on_second = float(data[node_path]['_golds_on_second'])
-				
+					load_golds_on_second(float(data[node_path]['_golds_on_second']))
 				"_xp_on_second":
-					game_data.xp_on_second = float(data[node_path]['_xp_on_second'])
-			
+					load_xp_on_second(float(data[node_path]['_xp_on_second']))
 				"_gold":
-					game_data.offline_gold_reward = float(data[node_path]['_golds_on_second']) * Offline_bonus_gold_ratio * game_data.offline_time
-					game_data.gold = float(data[node_path]['_gold']) + game_data.offline_gold_reward
-				
+					load_gold_and_reward(float(data[node_path]['_gold']), float(data[node_path]['_golds_on_second']))
 				"_xp":
-					game_data.offline_xp_reward = float(data[node_path]['_xp_on_second']) * Offline_bonus_xp_ratio * game_data.offline_time
-					game_data.xp = float(data[node_path]['_xp']) + game_data.offline_xp_reward
-				
+					load_xp_and_reward(float(data[node_path]['_xp']), float(data[node_path]['_xp_on_second']))
 				"_silver_moon":
-					game_data.silver_moon = int(data[node_path]['_silver_moon'])
-			
+					load_silver_moon(int(data[node_path]['_silver_moon']))
 				"_hp":
-					elf.set_current_hp(float(data[node_path]['_hp']))
-			
+					load_hp(float(data[node_path]['_hp']))
 				"_current_level":
-					var level_manager = get_parent().get_node("LevelManager")
-					level_manager.current_level = int(data[node_path]['_current_level'])
-					level_manager.set_level_label()
-					
+					load_level(int(data[node_path]['_current_level']))
 				"_elf_stats":
-					elf_stats.load_data(data[node_path]['_elf_stats'])
-					elf.reset_to_base()
-			
-				"_helth_potion":
-					get_parent().find_node("HealthPotion").load_data(data[node_path]['_helth_potion'])
-			
+					load_elf_stats(data[node_path]['_elf_stats'])
+				"_amount":
+					load_helth_potion(data[node_path]['_amount'])
 				"_price":
-					var items = get_parent().find_node("Items") 
-					for i in range(items.get_child_count()):
-						var item = items.get_child(i)	
-						var price = float(data[node_path]['_price'])
-						item.find_node("*BuyBtn").set_disabled(price > game_data.gold)
-						var ArrowDmgItem = get_parent().find_node("ArrowDmgItem")
-						ArrowDmgItem.price = price
-						ArrowDmgItem.update_price_label()
-	
-	game_data.update_gold_label()
-	game_data.update_xp_label()
-	game_data.update_silver_moon_label()
\ No newline at end of file
+					load_price(float(data[node_path]['_price']))
+	emit_signal("save_data_was_loaded")
+
+func load_offline_time(time):
+	var offline_time = OS.get_unix_time() - time
+	if OFFINE_LIMIT:
+		if offline_time > OFFINE_LIMIT_TIME:
+			offline_time = OFFINE_LIMIT_TIME
+	game_data.offline_time = offline_time
+
+func load_golds_on_second(gold_on_second):
+	game_data.golds_on_second = gold_on_second
+
+func load_xp_on_second(xp_on_second):
+	game_data.xp_on_second = xp_on_second
+
+func load_gold(gold):
+	game_data.gold = gold
+	#game_data.update_gold_label()
```

Comment:
We do not need that line, so we can remove it

Reply:
Nope, We need this func in revival_reset() but comments line is useless